utils/grecord.py

```diff
@@ -41,170 +46,213 @@ def __init__(self, parent):
+        self._pipeline.set_state(Gst.State.NULL)
         self._playing = False
 
     def is_playing(self):
         return self._playing
 
     def _get_state(self):
-        return self._pipeline.get_state()[1]
+        return self._pipeline.get_state(0)[1]
 
     def stop_recording_audio(self):
         # We should be able to simply pause and remove the audiobin, but
         # this seems to cause a gstreamer segfault. So we stop the whole
         # pipeline while manipulating it.
         # http://dev.laptop.org/ticket/10183
-        self._pipeline.set_state(gst.STATE_NULL)
+        self._pipeline.set_state(Gst.State.NULL)
         self._pipeline.remove(self._audiobin)
         self.play()
 
         audio_path = os.path.join(self._activity.datapath, 'output.wav')
         if not os.path.exists(audio_path) or os.path.getsize(audio_path) <= 0:
             # FIXME: inform model of failure?
+            _logger.error('output.wav does not exist or is empty')
             return
 
-        line = 'filesrc location=' + audio_path + ' name=audioFilesrc ! wavparse name=audioWavparse ! audioconvert name=audioAudioconvert ! vorbisenc name=audioVorbisenc ! oggmux name=audioOggmux ! filesink name=audioFilesink'
-        audioline = gst.parse_launch(line)
+        _logger.debug('stop_recording_audio')
+
+        line = 'filesrc location=' + audio_path + ' name=audioFilesrc ! \
+wavparse name=audioWavparse ! audioconvert name=audioAudioconvert ! \
+vorbisenc name=audioVorbisenc ! oggmux name=audioOggmux ! \
+filesink name=audioFilesink'
+        self._audioline = Gst.parse_launch(line)
 
-        vorbis_enc = audioline.get_by_name('audioVorbisenc')
+        vorbis_enc = self._audioline.get_by_name('audioVorbisenc')
 
-        audioFilesink = audioline.get_by_name('audioFilesink')
-        audioOggFilepath = os.path.join(self._activity.datapath,
-                                        'output.ogg')
-        audioFilesink.set_property("location", audioOggFilepath)
+        audioFilesink = self._audioline.get_by_name('audioFilesink')
+        audioOggFilepath = os.path.join(self._activity.datapath, 'output.ogg')
+        audioFilesink.set_property('location', audioOggFilepath)
 
-        audioBus = audioline.get_bus()
+        audioBus = self._audioline.get_bus()
         audioBus.add_signal_watch()
         self._audio_transcode_handler = audioBus.connect(
-            'message', self._onMuxedAudioMessageCb, audioline)
-        self._transcode_id = gobject.timeout_add(200, self._transcodeUpdateCb,
-                                                 audioline)
-        audioline.set_state(gst.STATE_PLAYING)
+            'message::eos', self._onMuxedAudioMessageCb, self._audioline)
+        self._transcode_id = GObject.timeout_add(200, self._transcodeUpdateCb,
+                                                 self._audioline)
+        self._audiopos = 0
+        self._audioline.set_state(Gst.State.PLAYING)
+
+    def transcoding_complete(self):
```

Comment:
Separate transcoding step was removed from the Record activity, so if you want to remove it here compare against those sources.  It was once needed because hardware could not record and transcode at the same time.